### ISSUE TITLE
overridden_super_call should recognize logic flow

### ISSUE THREAD

**Issue by andyjohns** [realm/SwiftLint#2667]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
An overridden function should not call super twice, however the logic flow of the function is not being taken in to account. If the function has two exit points, it may have the call to super in two places. For example:

```
    override func updateConstraints() {
        guard let item = itemModel else {
            super.updateConstraints()
            return
        }
        //.... do some logic .... 
        super.updateConstraints()
     }
```

The above will be flagged with the warning "Overridden methods call super Violation: Method 'updateConstraints()' should call to super only once (overridden_super_call)"

overridden_super_call false positive if the function has multiple exit points.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```
[Tue Mar 05 08:50:03 me:~/development/demo #] swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'ViewController.swift' (1/3)
Linting 'DemoView.swift' (2/3)
Linting 'AppDelegate.swift' (3/3)
/Users/andyjo/Development/demo/demo/DemoView.swift:21:40: warning: Overridden methods call super Violation: Method 'updateConstraints()' should call to super only once (overridden_super_call)
Done linting! Found 1 violation, 0 serious in 3 files.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.31.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?
installed SwiftLint.pkg, also CocoaPods in some projects

* Paste your configuration file:
```yml
# insert yaml contents here
```
disabled_rules:
    - line_length
    
opt_in_rules:
    - overridden_super_call

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
no
* Which Xcode version are you using (check `xcode-select -p`)?
10.1 (10B61)

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
    override func updateConstraints() {
        guard false else {
            super.updateConstraints()
            return
        }
        super.updateConstraints()
    }
```

**Comment by marcelofabri** [comment]
This is one of the reasons this rule is opt-in. We probably could catch the case when a `guard` is used, but it's hard to detect the general case (e.g. in more complex branches)
